packages/aws-cdk-lib/aws-ecs/lib/base/base-service.ts

```diff
@@ -721,6 +740,46 @@ export abstract class BaseService extends Resource
     this.node.defaultChild = this.resource;
   }
 
+  /**
+   * Adds a volume to the Service.
+   */
+  public addVolume(volume: ServiceManagedVolume) {
+    this.volumes.push(volume);
+  }
+
+  private renderVolumes(): CfnService.ServiceVolumeConfigurationProperty[] {
```

Comment:
Since only one EBS configuration should be set, it might be a good to check here whether the volumes element is one or not.

CloudFormation returns the error `Invalid request provided: UpdateService error: The provided volume configuration is invalid. Only one volume can be configured at launch.`.

Reply:
added a validation check to catch this scenario over here [[link](https://github.com/aws/aws-cdk/pull/28691/files#diff-95d9ac7a97b34b5521d19daaf5464f6dc5de93fd10abab0add914683dd430ff7R747-R750)]